### PR TITLE
fix: correct help for subcommands

### LINKED ISSUE #171: command help is broken
`mix help` and `mix projects help` work as expected but `mix projects:list help` fails.

```
$ mix projects:replace --help
replace the content of a project

    TypeError: (cmd.args || []).filter is not a function

```

It appears that oclif v2 and its most recent help plugin expect to find a defined `args` property on the command class when help is requested for a specific command like `projects:list`.

mix-cli makes exclusive use of flags so arguments were never defined for any command. Just to be clear, an "argument" is an input you pass to the cli without a flag identifying it. They are often called "ordered arguments" since you cannot provided them in any order like you can with flags.

The solution to this should be to simply define an `args` class intances in `base-command.ts` and set it to `[]`.

### PR DESCRIPTION
**Pull request by TheTrio**
Resolves #171 

Have tested it out using the two tar files generated by `npm pack`. The one generated on `main` gives the TypeError. The one generated on this branch doesn't have that error.

### CODE REVIEW

**Comment by grof** [comment]
I have also tried it on my side, the problem is fixed. The only thing missing is a unit test to make sure help does not get broken in the future.  We are going to add unit to make sure the following commands still work:

- mix --help  => check for presence of one topic and one command in the help output
- mix projects --help  => check for presence of one command in the help output
- mix projects:list --help => check for presence of one flag description.

Let's put those tests in a new file at test/help.test.ts

**Comment by TheTrio** [comment]
Have added the tests in

File: test/help.test.ts [hunk 1]
```diff
@@ -0,0 +1,35 @@
+
+import {expect, test} from '@oclif/test'
+
+describe('Help message handling', () => {
+  test
+    .stdout()
+    .command(['help'])
+  .it('works for global Mix', (ctx)=>{
```
Comment (grof) [inline test/help.test.ts:8]: It's preferrable if a unit test description provides the complete context. That way, if a unit test fails, the unit test description of the failed test provides complete information.

Update the unit test description to "provides global help".

File: test/help.test.ts [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    expect(ctx.stdout).to.include('obtain Mix access token')
+  })
+
+  // we use the help subcommand instead of passing in --help to the projects:list command
+  // this is to workaround the weird behavior of oclif where it will not accept --help
+  // during testing, even though it works fine when running the command normally
+  test
+    .stdout()
+    .command(['help', 'projects:list'])
+  .it('works for projects:list', (ctx)=>{
```
Comment (grof) [inline test/help.test.ts:21]: Update the unit test description to "provides help for a specific command".

**Review by grof — APPROVED** [review]

File: test/help.test.ts [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    expect(ctx.stdout).to.include('--json')
+    expect(ctx.stdout).to.include('list projects')
+    expect(ctx.stdout).to.include('exclude project channels from the list')
+  })
+
+  test
+    .stdout()
+    .command(['help', 'projects'])
+  .it('works for projects', (ctx)=>{
```
[the remark this answers is not in the document]
Comment (grof) [reply test/help.test.ts]: In the documenation and design docs, "command" is used to refer to a combination of <topic>:<action> and "topic" is use to refer to just <topic>. I made the changes to reflect that.

**Pull request merged**